### PR TITLE
DIG-590: Fix sms code retrieval time tracking on fragment change

### PR DESCRIPTION
**Pull request by MelanijaGrunte**

### CODE REVIEW

File: app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt [hunk 1]
```diff
@@ -30,13 +31,17 @@ class SmsViewModel @Inject constructor(): BaseViewModel() {
         _smsRetrievalEventLiveData.value = Event(true)
     }
 
-    fun startCodeRetrievalTimer(interval: Long = 1000, showMessage: Boolean = false) {
+    fun startCodeRetrievalTimer(interval: Long = 0L, showMessage: Boolean = false) {
         val timeRemainingMillis = DateTime.now().millis - (lastCodeRequestTime)
-        if (timeRemainingMillis < interval) {
+        if (interval == 0L && lastCodeRequestTime == 0L) {
```
Comment (ULazdins) [inline app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt:36]: I have a feeling that we actually have two big sub-cases here - user can send SMS and she cant. So split the code accordingly :)

[the remark this answers is not in the document]
Comment (ULazdins) [reply app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt:36]: Also there should be no difference if `lastCodeRequestTime` is 0 or an hour ago. Both cases mean that the user can resend SMS

File: app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt [hunk 2]
```diff
@@ -30,13 +39,21 @@ class SmsViewModel @Inject constructor(): BaseViewModel() {
...
-        if (timeRemainingMillis < interval) {
-            _requestRemainingTimeShowMessagePairLiveData.value = Pair(timeRemainingMillis - interval, showMessage)
+    fun processSmsRequest() {
+        val timeRemainingMillis = DateTime.now().millis - sharedPreferenceManager.lastSmsCodeRequestTime
+        if (timeRemainingMillis >= MILLIS_IN_MINUTE) {
+            _sendSmsWithMessageLiveData.value = Event(false)
+        }
+    }
+
+    fun processSmsRequestAndStartTimer() {
```
Comment (mkims) [inline app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt:49]: `..AndLogTimeIfNeeded`? Kaut kā tā? Timera jau nav nekāda

File: app/src/main/java/lv/spkc/apturicovid/ui/sms/SmsRetrievalFragment.kt [hunk 3]
```diff
@@ -73,11 +69,17 @@ class SmsRetrievalFragment: BaseFragment() {
...
+        observeEvent(smsViewModel.smsRemainingRequestTimeLiveData) {
+            val errorText = String.format(getString(R.string.label_code_resend_error), it)
+            Toast.makeText(requireContext(), errorText, Toast.LENGTH_SHORT).show()
         }
 
-        observeLiveData(smsViewModel.requestRemainingTimeShowMessagePairLiveData) {
-            handleSendNumberRequest(it.first, it.second)
+        observeEvent(smsViewModel.sendSmsWithMessageLiveData) {
+            sendNumber()
+            if (it) Toast.makeText(requireContext(), getString(R.string.label_code_resent), Toast.LENGTH_SHORT).show()
```
Comment (mkims) [inline app/src/main/java/lv/spkc/apturicovid/ui/sms/SmsRetrievalFragment.kt:82]: P.S. pēc if {} vajag

File: app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt [hunk 4]
```diff
@@ -5,20 +5,29 @@ import androidx.lifecycle.MutableLiveData
...
     val smsRetrievalEventLiveData: LiveData<Event<Boolean>> = _smsRetrievalEventLiveData
 
-    private var lastCodeRequestTime: Long = 0L
+    private val _smsRemainingRequestTimeLiveData = MutableLiveData<Event<Long>>()
+    val smsRemainingRequestTimeLiveData: LiveData<Event<Long>> = _smsRemainingRequestTimeLiveData
 
-    private val _requestRemainingTimeShowMessagePairLiveData = MutableLiveData<Pair<Long, Boolean>>()
-    val requestRemainingTimeShowMessagePairLiveData: LiveData<Pair<Long, Boolean>> = _requestRemainingTimeShowMessagePairLiveData
+    private val _sendSmsWithMessageLiveData = MutableLiveData<Event<Boolean>>()
+    val sendSmsWithMessageLiveData: LiveData<Event<Boolean>> = _sendSmsWithMessageLiveData
```
Comment (mkims) [inline app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt:30]: Te mazliet wording mūlsinā.
maybe `AndShowMessage?` 
un komentāru, ka true/false ietekmē toastu, nevis sms sūtīšanu

File: app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt [hunk 5]
```diff
@@ -30,13 +39,21 @@ class SmsViewModel @Inject constructor(): BaseViewModel() {
...
+        val timeRemainingMillis = DateTime.now().millis - sharedPreferenceManager.lastSmsCodeRequestTime
+        if (timeRemainingMillis >= MILLIS_IN_MINUTE) {
+            _sendSmsWithMessageLiveData.value = Event(false)
+        }
+    }
+
+    fun processSmsRequestAndStartTimer() {
+        val timeRemainingMillis = DateTime.now().millis - sharedPreferenceManager.lastSmsCodeRequestTime
+        if (timeRemainingMillis < MILLIS_IN_MINUTE) {
+            val remainingTime = (timeRemainingMillis - MILLIS_IN_MINUTE) * -1 / MILLIS_IN_SECOND
```
Comment (mkims) [inline app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt:52]: `MILLIS_IN_MINUTE - timeRemainingMillis`? un bez *-1

**Pull request merged**